src/OpenLoco/IndustryManager.cpp

```diff
@@ -190,18 +190,238 @@ namespace OpenLoco::IndustryManager
         }
     }
 
+    // 0x00459A05
+    static bool isTooCloseToNearbyIndustries(const Map::Pos2& loc)
+    {
+        for (auto& industry : industries())
+        {
+            const auto dist = Math::Vector::manhattanDistance(loc, Map::Pos2{ industry.x, industry.y });
+            if (dist < 480)
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    // 0x00459A50
+    static bool isOutwithCluster(const Map::Pos2& loc, const uint8_t indObjId)
+    {
+        auto numClusters = 0;
+        for (auto& industry : industries())
+        {
+            if (industry.objectId != indObjId)
+            {
+                continue;
+            }
+            numClusters++;
+            const auto dist = Math::Vector::manhattanDistance(loc, Map::Pos2{ industry.x, industry.y });
+            if (dist < 960)
+            {
+                return false;
+            }
+        }
+        if (numClusters < 3)
+        {
+            return false;
+        }
+        return true;
+    }
+
+    // 0x00469A81
+    static int16_t mountainHeight(const Map::Pos2& loc)
+    {
+        // Works out roughly the height of a mountain of area 11 * 11
+        // (Its just the heighest point - the lowest point)
+        int16_t lowest = std::numeric_limits<int16_t>::max();
+        int16_t highest = 0;
+        Map::TilePosRangeView range{
+            loc - Map::TilePos2{ 5, 5 }, loc + Map::TilePos2{ 5, 5 }
+        };
+        for (auto& tilePos : range)
+        {
+            if (!Map::validCoords(tilePos))
+            {
+                continue;
+            }
+            auto tile = Map::TileManager::get(tilePos);
+            auto* surface = tile.surface();
+            auto height = surface->baseHeight();
+            lowest = std::min(lowest, height);
+            if (surface->slope())
+            {
+                height += 16;
+                if (surface->isSlopeDoubleHeight())
+                {
+                    height += 16;
+                }
+            }
+            highest = std::max(highest, height);
+        }
+        return highest - lowest;
+    }
+
     // 0x004599B3
     static std::optional<Map::Pos2> findRandomNewIndustryLocation(const uint8_t indObjId)
     {
-        registers regs;
-        regs.edx = indObjId;
-        call(0x004599B3, regs);
-        if (regs.ax == -1)
+        auto* indObj = ObjectManager::get<IndustryObject>(indObjId);
+        for (auto i = 0; i < 250; ++i)
         {
-            return std::nullopt;
-        }
+            // Replace the below with this after validating the function
```

Comment:
Is this intended for another PR, or should this be done at this point? I realise this would break our replays.

Reply:
Its intended for after doing all industry manager functions. Probably after the map gen is done.